File: src/main/java/com/wowo/wowo/domain/transaction/valueobjects/TransactionType.java
package com.wowo.wowo.domain.transaction.valueobjects;

import lombok.Getter;

/**
 * Enumeration representing the type of transaction flow.
 */
@Getter
public enum TransactionType {
    TRANSFER_MONEY("Transfer Money"),
    TOP_UP("Top Up"),
    WITHDRAW("Withdraw"),
    PAYMENT("Payment"),
    REFUND("Refund"),
    DEPOSIT("Deposit");
    
    private final String description;
    
    TransactionType(String description) {
        this.description = description;
    }
}

File: src/main/kotlin/com/wowo/wowo/contexts/transaction/domain/valueobject/TransactionType.kt
package com.wowo.wowo.contexts.transaction.domain.valueobject

import com.wowo.wowo.shared.domain.ValueObject

/**
 * Value Object representing transaction type
 */
enum class TransactionType : ValueObject {
    CREDIT,     // Money coming in
    DEBIT,      // Money going out
    TRANSFER,   // Transfer between wallets
    PAYMENT,    // Payment to merchant
    REFUND      // Refund from merchant
}

